package.go

```diff
@@ -75,7 +74,7 @@ func ImportsForType(currPkg string, t reflect.Type) []Import {
 		return dedupImports(append(ImportsForType(currPkg, t.Key()), ImportsForType(currPkg, t.Elem())...))
 	default:
 		path := t.PkgPath()
-		if path == "" || path == currPkg {
+		if path == "" || path == currPkg || path == "main" {
```

Comment:
It comes in from gen/main.go when you have to insert a placeholder/dummy type in place of a generic, so you end up importing `"main"` because it thinks that's a relevant package while we also strip it out here. Perhaps there's a more intelligent way to do it here but it'll require more wiring to figure out the packages of the placeholders and pass it down here. I can work on that if you think it's important but my assumption here is that gen will (almost) always be done from a `"main"` package and those dummy packages will be defined in there alongside.

Reply:
Ah... I see. And _we_ can't supply the dummy types because there may be additional constraints. That's probably fine for now.

In the future, we can probably use `runtime.Caller` get the file that's generating the code.

Reply:
actually this was easily solvable in the caller of this function:

```go
		if gti.toGenericTypeParam(f.Type.String()) == "" {
			imports = append(imports, ImportsForType(f.Pkg, f.Type)...)
		}
```

We can look at `f.Type` to see if it's one of the generic placeholders